### PR TITLE
Fix: bugs during internal bugathon and fix the alias deletion issue in security mode.

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/Aliases/containers/DeleteAliasModal/DeleteAliasModal.tsx [hunk 1]
```diff
@@ -44,14 +45,25 @@ export default function DeleteAliasModal(props: DeleteAliasModalProps) {
   const onConfirm = useCallback(async () => {
     if (services) {
       const result = await services.commonService.apiCaller({
-        endpoint: "indices.deleteAlias",
+        endpoint: "indices.updateAliases",
```
Comment (gaobinlong) [inline public/pages/Aliases/containers/DeleteAliasModal/DeleteAliasModal.tsx:48]: Why not using delete alias API and specifying concrete index names to delete all of the aliases, maybe simpler than this?

File: public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ export default function IndexDetail(props: IndexDetailModalProps) {
     const result = await services.indexService.getIndices({
       terms: index,
       from: 0,
-      size: 10,
+      size: 999,
```
Comment (Hailong-am) [inline public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx:86]: any reason changed it to 999?

File: public/pages/Aliases/containers/DeleteAliasModal/DeleteAliasModal.tsx [hunk 3]
```diff
@@ -63,7 +75,7 @@ export default function DeleteAliasModal(props: DeleteAliasModalProps) {
     return null;
   }
 
-  const hasSystemIndex = props.selectedItems.some((index) => filterByMinimatch(index, SYSTEM_ALIAS));
+  const hasSystemIndex = props.selectedItems.some((index) => filterByMinimatch(index.alias, SYSTEM_ALIAS));
```
Comment (xluo-aws) [inline public/pages/Aliases/containers/DeleteAliasModal/DeleteAliasModal.tsx:78]: Is this a different bug?

File: public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx (hunk 2 above)
Comment (SuZhou-Joe) [reply public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx:86]: The getIndices API fetch indexes by using "like", and if there are more than 10 indexes matched, the target Index can not be found.

File: public/pages/Aliases/containers/DeleteAliasModal/DeleteAliasModal.tsx (hunk 3 above)
Comment (SuZhou-Joe) [reply public/pages/Aliases/containers/DeleteAliasModal/DeleteAliasModal.tsx:78]: Nope, the selectedItems in the props changed, it was a string[] but now it's a IAlias[].

File: public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx (hunk 2 above)
Comment (SuZhou-Joe) [reply public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx]: But good point, I will change to add exactSearch on getIndices method and lower the size to 10.

**Review by Hailong-am — APPROVED** [review]

**Review by gaobinlong — APPROVED** [review]

**Pull request merged**